### PR TITLE
Create the Tickets JSON-LD

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#43595](https://central.tri.be/issues/43595)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $name is never used | [Ticket.php:31](https://github.com/moderntribe/event-tickets/blob/e262734e9ad70c8bd12aa83b5486cc6bab67ff4e/src/Tribe/JSON_LD/Ticket.php#L31)
:warning: | The method parameter $args is never used | [Main.php:247](https://github.com/moderntribe/event-tickets/blob/e262734e9ad70c8bd12aa83b5486cc6bab67ff4e/src/Tribe/Main.php#L247)

> via jenkins/codesniffer-phpcs

File: src/Tribe/JSON_LD/Ticket.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	 */
+	public $type = 'Offer';
+
+	/**
+	 * On PHP 5.2 the child class doesn't get spawned on the Parent one, so we don't have
+	 * access to that information on the other side unless we pass it around as a param
+	 * so we throw __CLASS__ to the parent::instance() method to be able to spawn new instance
+	 * of this class and save on the parent::$instances variable.
+	 *
+	 * @param   $name DONT USE THIS PARAM, it's here for Abstract compatibility
```
Comment (barryhughes) [inline src/Tribe/JSON_LD/Ticket.php]: Spacing.

File: src/Tribe/JSON_LD/Ticket.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	/**
+	 * Which type of element this actually is
+	 *
+	 * @see https://developers.google.com/structured-data/rich-snippets/
+	 * @var string
+	 */
+	public $type = 'Offer';
+
+	/**
+	 * On PHP 5.2 the child class doesn't get spawned on the Parent one, so we don't have
```
Comment (barryhughes) [inline src/Tribe/JSON_LD/Ticket.php]: I could be misunderstanding ... but if we used `get_class( $this )` (in the superclass) mightn't that clean up the do-not-use-param in the method signatures of the subclasses in a 5.2-compatible way?

File: src/Tribe/JSON_LD/Ticket.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	 *
+	 * @param  int|WP_Post|null $post The post/event
+	 * @param  array  $args
+	 * @return array
+	 */
+	public function get_data( $tickets = null, $args = array() ) {
+		$tickets = (array) $tickets;
+		$return = array();
+
+		foreach ( $tickets as $i => $post ) {
```
Comment (barryhughes) [inline src/Tribe/JSON_LD/Ticket.php]: We're not using `$i` - can we ditch it?

File: src/Tribe/Main.php [hunk 4]
```diff
@@ -243,6 +244,45 @@ public function hooks() {
...
+
+		if ( empty( $tickets ) ) {
+			return $data;
+		}
+
+		$data->offers = array();
+		$startdates = array();
+		$enddates = array();
+		$in_stock = array();
+		foreach ( $tickets as $i => $ticket ) {
```
Comment (barryhughes) [inline src/Tribe/Main.php]: Similar to the above, could drop `$i` here?

**Comment by bordoni** [comment]
Updated :+1:

File: src/Tribe/Main.php [hunk 5]
```diff
@@ -243,6 +244,45 @@ public function hooks() {
 		}
 	}
 
+	public function inject_tickets_json_ld( $data, $args, $post ) {
```
Comment (borkweb) [inline src/Tribe/Main.php]: I'd love to see a docblock here that at the minimum documents the parameters.

**Comment by bordoni** [comment]
Updated

**Comment by borkweb** [comment]
Beloved. Hahaha. :+1:

**Pull request merged**